Fixes #1189: ProfileChooser-FirstTime&Regular - Tablet (Landscape) (Highfi)

MohamedMedhat1998 opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->
## Note
The branch name should be `profile-chooser-ft-tablet-land-highfi` not `profile-chooser-ft-tablet-land-lowfi`, **highfi** not **lowfi**.  
Sorry about that mistake.
## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
- Implemented High-fi UI for ProfileChooser-FirstTime & ProfileChooser-Regular tablet landscape
- Mock: https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/33741ea7-89da-4e5d-b2a4-9872f6db48b4/
- Fixes #1189 
- Fixes #1137 
- Fixes #1138 
## Screenshots (1024x768) emulator

![screenshot-2020-06-15_14 14 41 943](https://user-images.githubusercontent.com/8533363/84657141-1e40ee80-af14-11ea-8235-0ac969591164.png)
![screenshot-2020-06-15_14 16 53 798](https://user-images.githubusercontent.com/8533363/84657145-1f721b80-af14-11ea-8b11-c45a386bad9d.png)
![screenshot-2020-06-15_14 17 22 518](https://user-images.githubusercontent.com/8533363/84657147-200ab200-af14-11ea-8c99-3d0897f0c62a.png)
![screenshot-2020-06-15_14 19 14 926](https://user-images.githubusercontent.com/8533363/84657149-21d47580-af14-11ea-9c60-5a4edbc8da05.png)
![screenshot-2020-06-15_14 19 23 869](https://user-images.githubusercontent.com/8533363/84657151-2305a280-af14-11ea-8b60-2550c38fe969.png)

## Screenshots (2560x1800) emulator

![screenshot-2020-06-15_14 09 39 519](https://user-images.githubusercontent.com/8533363/84657328-695b0180-af14-11ea-94b0-12228c9eab07.png)
![screenshot-2020-06-15_14 05 27 984](https://user-images.githubusercontent.com/8533363/84657330-6a8c2e80-af14-11ea-8614-2d3124f8d0e9.png)
![screenshot-2020-06-15_14 07 49 951](https://user-images.githubusercontent.com/8533363/84657321-66f8a780-af14-11ea-89af-148cb98e2f9f.png)

### Update
Updated the max width of `add_profile_description_text` `TextView` so that the sentence appears in two lines instead of one line.
![screenshot-2020-06-16_18 49 25 233](https://user-images.githubusercontent.com/8533363/84803680-9ab1fb00-b002-11ea-9319-a6997bb1a4b7.png)


## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

MohamedMedhat1998 committed:
-Completed "ProfileChooser-FirstTime" tablet (port) (land) low-fi

MohamedMedhat1998 committed:
-Added EOF in "booleans.xml"

MohamedMedhat1998 committed:
Merge remote-tracking branch 'origin/develop' into profile-chooser-ft-tablet-land-lowfi

MohamedMedhat1998 committed:
-Restored the lost work

rt4914 reviewed (changes requested):
@MohamedMedhat1998 As discussed in the chat, these PRs should be done like this:
PR 1 - Entire landscape mode for regular + first-time.
PR 2 - Entire portrait mode for regular + first-time.


This is needed because of the interdependency of `first-time` and `regular` code and it is not possible to review half part because even if we do that in later stage again we will need to review both parts because there might be some changes. 

So I suggest including the `regular` view for landscape mode in this PR to make things work correctly.

Thanks.

MohamedMedhat1998 commented:
> 
> 
> @MohamedMedhat1998 As discussed in the chat, these PRs should be done like this:
> PR 1 - Entire landscape mode for regular + first-time.
> PR 2 - Entire portrait mode for regular + first-time.
> 
> This is needed because of the interdependency of `first-time` and `regular` code and it is not possible to review half part because even if we do that in later stage again we will need to review both parts because there might be some changes.
> 
> So I suggest including the `regular` view for landscape mode in this PR to make things work correctly.
> 
> Thanks.

@rt4914 Ok, I'll be working on it asap.  
Thanks!

MohamedMedhat1998 committed:
-Applied Pulkit's suggestion which is using "DividerItemDecoration"

MohamedMedhat1998 committed:
-Updated the top margins of the recycler view in the profile chooser

rt4914 commented on app/src/main/res/layout-sw600dp-land/profile_chooser_add_view.xml:
```diff
@@ -0,0 +1,86 @@
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:layout_marginTop="@dimen/margin_0"
+      android:paddingStart="@{presenter.wasProfileEverBeenAddedValue ? @dimen/margin_0 : @dimen/margin_24}"
```
As per the mocks there should be marginEnd = 24dp too when first time chooser view is show. 

This is also make sure that the description text in looks good in 2 lines in this image:
https://user-images.githubusercontent.com/8533363/84657328-695b0180-af14-11ea-94b0-12228c9eab07.png

MohamedMedhat1998 replied:
Done!

rt4914 commented on app/src/main/res/layout-sw600dp-land/profile_chooser_profile_view.xml:
```diff
@@ -0,0 +1,108 @@
+    <LinearLayout
+      android:layout_width="wrap_content"
+      android:layout_height="wrap_content"
+      android:paddingStart="@{presenter.wasProfileEverBeenAddedValue ? @dimen/margin_0 : @dimen/margin_24}"
```
Ditto as `add_view` comment.

MohamedMedhat1998 replied:
Done!

rt4914 reviewed (changes requested):
LGTM, one nit change suggested.

MohamedMedhat1998 commented:
@mschanteltc Hi Chantel. Please take a look at the screenshots here that contain 5 items in a row.  
They are from a larger emulator (2560x1800), so it will be better to make the row take 5 items instead of 4.  
Rajat reviewed these screens and approved them visually. We are just waiting for your approval.  
Do they look good or there should be some change to be made?  
Thanks!

MohamedMedhat1998 committed:
-Updated the max width of the "add_profile_description_text" textView

MohamedMedhat1998 committed:
Merge remote-tracking branch 'origin/develop' into profile-chooser-ft-tablet-land-lowfi

# Conflicts: (resolved)
#	app/src/main/java/org/oppia/app/profile/ProfileChooserFragmentPresenter.kt

mschanteltc commented:
> @mschanteltc Hi Chantel. Please take a look at the screenshots here that contain 5 items in a row.
> They are from a larger emulator (2560x1800), so it will be better to make the row take 5 items instead of 4.
> Rajat reviewed these screens and approved them visually. We are just waiting for your approval.
> Do they look good or there should be some change to be made?
> Thanks!

These look great to me, awesome work Mohamed!

rt4914 commented on app/src/main/res/layout-sw600dp-land/profile_chooser_add_view.xml:
```diff
@@ -0,0 +1,89 @@
+      android:layout_height="wrap_content"
+      android:layout_marginTop="@dimen/margin_0"
+      android:paddingStart="@{presenter.wasProfileEverBeenAddedValue ? @dimen/margin_0 : @dimen/margin_24}"
+      android:paddingEnd="@{presenter.wasProfileEverBeenAddedValue ? @dimen/margin_0 : @dimen/margin_24}"
```
There is one thing that I missed las time, as you are using padding, your dimen values should be `padding_24` or you should use `marginStart` and `margin_24`.
I think you should use `margins` here instead of `padding`

rt4914 replied:
Ditto for other file.

MohamedMedhat1998 replied:
Done

rt4914 commented on app/src/main/res/layout-sw600dp-land/profile_chooser_fragment.xml:
```diff
@@ -0,0 +1,134 @@
+          android:paddingBottom="@{presenter.wasProfileEverBeenAddedValue ? @dimen/padding_296dp : @dimen/bottom_white_space}"
+          android:requiresFadingEdge="vertical"
+          android:scrollbars="none"
+          android:tag="profile_recycler_view"
```
Remove this tag

MohamedMedhat1998 replied:
Done

rt4914 reviewed (changes requested):
Some final nit changes.

MohamedMedhat1998 committed:
-Replaced "padding" with "margins" in "profile_chooser_add_view" and "profile_chooser_profile_view"
-Removed the tag in "profile_chooser_fragment"
-Merged with the develop and resolved the conflict

rt4914 reviewed (approved):
LGTM, thanks.
